conanfile.py

```diff
@@ -67,10 +67,19 @@ def make_requirements_file(self, extra_packages=None):
         The `extra_packages` can be used to add extra packages (as a Python `list`) to be 
         installed in addition to `self.options.packages`.
         """
-        packages_str = str(self.options.packages)
-        is_file = "\n" in packages_str  # requirements.txt as opposed to space-separated list
-        packages_list = packages_str.strip().split("\n" if is_file else " ")
-
+        def split_lines(string):
+            """`options.packages` may be encoded as tab, newline or space separated
+
+            The `\n` separator doesn't play well with Conan but we need to support 
+            it for backward compatibility.
+            """
+            for separator in ["\t", "\n"]:
```

Comment:
If we require the `.replace("\n", "\t")` modification when reading in files, do we need to check for `"\n"` here as a separator? Or am I misunderstanding the PR description?

Reply:
One more thought, would it make sense to mention the need for the `.replace("\n", "\t")` calls in case of comments somewhere in the `conanfile.py`?

Reply:
Yes. I have another branch with a full example project and instructions. I'll open a PR for it.